services/cancellation.ts

```diff
@@ -58,7 +58,12 @@ class CancellationService implements ICancellationService {
 	}
 
 	private static get killSwitchDir(): string {
-		return path.join(os.tmpdir(), process.env.SUDO_USER || process.env.USER || process.env.USERNAME, "KillSwitches");
+        var tmp = process.env.SUDO_USER || process.env.USER || process.env.USERNAME;
+        if (tmp) {
+            return path.join(os.tmpdir(), tmp, "KillSwitches");
+        } else {
+            return path.join(os.tmpdir(), "KillSwitches");
+        }
```

Comment:
Interesting.   I just tested path.join and your understanding is correct.  I had expected path join to do two slashes in the middle of the path...   I'll repush a commit.